services/mock-service/src/fsp-integration/onafriq/onafriq.mock.service.ts

```diff
@@ -0,0 +1,126 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+import { setTimeout } from 'node:timers/promises';
+import { lastValueFrom } from 'rxjs';
+
+import { API_PATHS, EXTERNAL_API_ROOT } from '@mock-service/src/config';
+import {
+  OnafriqCallServicePayload,
+  OnafriqCallServiceResponseBodyDto,
+} from '@mock-service/src/fsp-integration/onafriq/onafriq.dto';
+
+enum MockScenario {
+  success = 'success',
+  errorOnRequest = 'error-on-request',
+  errorOnCallback = 'error-on-callback',
+}
+@Injectable()
+export class OnafriqMockService {
+  public async callService(
+    callServiceDto: OnafriqCallServicePayload,
+  ): Promise<OnafriqCallServiceResponseBodyDto> {
+    let mockScenario: MockScenario = MockScenario.success;
+    if (callServiceDto.requestBody[0].recipient.msisdn === '24300000000') {
+      mockScenario = MockScenario.errorOnRequest;
+    } else if (
+      callServiceDto.requestBody[0].recipient.msisdn === '24300000001'
+    ) {
+      mockScenario = MockScenario.errorOnCallback;
+    }
+
+    if (mockScenario === MockScenario.errorOnRequest) {
+      return {
+        totalTxSent: 1,
+        noTxAccepted: 0,
+        noTxRejected: 1,
+        details: {
+          transResponse: [
+            {
+              thirdPartyId: callServiceDto.requestBody[0].thirdPartyTransId,
+              status: {
+                code: '101',
+                message: 'Rejected',
+                messageDetail: 'Generic mock error on request',
+              },
+            },
+          ],
+        },
+        timestamp: new Date().toISOString(),
+      };
+    }
+
+    const callServiceResponse = {
+      totalTxSent: 1,
+      noTxAccepted: 1,
+      noTxRejected: 0,
+      details: {
+        transResponse: [
+          {
+            thirdPartyId: callServiceDto.requestBody[0].thirdPartyTransId,
+            status: {
+              code: '100',
+              message: 'Accepted',
+            },
+          },
+        ],
+      },
+      timestamp: new Date().toISOString(),
+    };
```

Comment:
I think making the responses here contrains a bit of duplicate code. Would be cleaner to have a function or something to create the responses.